lms/djangoapps/certificates/generation.py

```diff
@@ -56,10 +57,13 @@ def generate_course_certificate(user, course_key, generation_mode):
         emit_certificate_event(event_name='created', user=user, course_id=course_key, event_data=event_data)
         emit_segment_event(user_id=user.id, course_id=course_key)
 
+    elif CertificateStatuses.unverified == cert.status:
+        cert.mark_unverified(source='certificate_generation')
```

Comment:
I feel like I may be missing something. If the status is `unverified` and gets set during the `_generate_certificate` function call, why are calling `mark_unverified` afterward? Is it just to ensure we are calling all the signals that may be needed in our revoke code path?

Reply:
Yep, that's exactly it (to ensure the signals are sent). I'm open to alternate approaches/suggestions.

Reply:
It just redundant to me at first, but then I assumed it was related to the signals. I figured I'd ask to check my assumption. :)